srcpkgs/xrootd/template

```diff
@@ -0,0 +1,13 @@
+# Template file for 'xrootd'
+pkgname=xrootd
+version=5.4.0
+revision=1
+archs="~*-musl"
```

Comment:
It seems it's more complicated than that, and make take a bigger effort. After patching that, I also had to change a few `__GNUC__` s to `__GLIBC__`, and am still running into random compilation errors like `error: invalid conversion from 'int*' to 'socklen_t*'`. `xrootd` might be a little too tied to glibc.

Reply:
As for the `root-R` issue, it's because of one of `R-cran-Rcpp`'s header files the R bindings need to compile uses `execinfo.h`, which musl doesn't have.

Reply:
R-cran-Rcpp is fixed by #35301

Reply:
Great, the `root-R` part compiles on musl now without issue, so I removed the restriction.

Reply:
Ok, so I stumbled through throwing together a patch for xrootd to work on musl and pushed that here. I compiled and briefly tested the patched version on glibc and musl (on amd64). Someone let me know if anything looks clearly off. I could also send it upstream.